clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp

```diff
@@ -1458,6 +1520,243 @@ inline void csr2csc(sycl::queue queue, int m, int n, int nnz,
+    int cols_a = (trans_a == oneapi::mkl::transpose::nontrans) ? k : m;
+    int rows_b = (trans_b == oneapi::mkl::transpose::nontrans) ? k : n;
+    int cols_b = (trans_b == oneapi::mkl::transpose::nontrans) ? n : k;
+
+#ifdef DPCT_USM_LEVEL_NONE
+    oneapi::mkl::sparse::set_csr_data(queue, *matrix_handle_a, rows_a, cols_a,
+                                      info_a->get_index_base(), row_ptr_a_buf,
+                                      col_ind_a_buf, val_a_buf);
+    oneapi::mkl::sparse::set_csr_data(queue, *matrix_handle_b, rows_b, cols_b,
+                                      info_b->get_index_base(), row_ptr_b_buf,
+                                      col_ind_b_buf, val_b_buf);
+    oneapi::mkl::sparse::set_csr_data(queue, *matrix_handle_c, rows_c, cols_c,
+                                      info_c->get_index_base(), row_ptr_c_buf,
+                                      col_ind_c_buf, val_c_buf);
+#else
+    oneapi::mkl::sparse::set_csr_data(
+        queue, *matrix_handle_a, rows_a, cols_a, info_a->get_index_base(),
+        (int *)row_ptr_a, (int *)col_ind_a, (Ty *)val_a);
+    oneapi::mkl::sparse::set_csr_data(
+        queue, *matrix_handle_b, rows_b, cols_b, info_b->get_index_base(),
+        (int *)row_ptr_b, (int *)col_ind_b, (Ty *)val_b);
+    oneapi::mkl::sparse::set_csr_data(queue, *matrix_handle_c, rows_c, cols_c,
+                                      info_c->get_index_base(), row_ptr_c,
+                                      (int *)nullptr, (Ty *)nullptr);
+#endif
+
+    matmat_desc = new oneapi::mkl::sparse::matmat_descr_t();
+    oneapi::mkl::sparse::init_matmat_descr(matmat_desc);
+    oneapi::mkl::sparse::set_matmat_data(
+        *matmat_desc, oneapi::mkl::sparse::matrix_view_descr::general, trans_a,
+        oneapi::mkl::sparse::matrix_view_descr::general, trans_b,
+        oneapi::mkl::sparse::matrix_view_descr::general);
+    desc->get_csrgemm_info_map().insert(std::make_pair(
+        args, descriptor::matmat_info(matrix_handle_a, matrix_handle_b,
+                                      nullptr, matrix_handle_c, matmat_desc)));
+  } else {
+    matrix_handle_a = Iter->second.matrix_handle_a;
+    matrix_handle_b = Iter->second.matrix_handle_b;
+    matrix_handle_c = Iter->second.matrix_handle_c;
+    matmat_desc = Iter->second.matmat_desc;
+  }
+
+#ifdef DPCT_USM_LEVEL_NONE
+#define __MATMAT(STEP, NNZ_C)                                                  \
+  oneapi::mkl::sparse::matmat(queue, *matrix_handle_a, *matrix_handle_b,       \
+                              *matrix_handle_c, STEP, *matmat_desc, NNZ_C,     \
+                              nullptr)
+#else
+#define __MATMAT(STEP, NNZ_C)                                                  \
+  oneapi::mkl::sparse::matmat(queue, *matrix_handle_a, *matrix_handle_b,       \
+                              *matrix_handle_c, STEP, *matmat_desc, NNZ_C,     \
+                              nullptr, {})
+#endif
```

Comment:
It looks like SYCLomatic is going to migrate `cusparse<t>csrgemm` APIs in a way that oneMKL will allocate whatever temporary workspace it needs to internally, i.e., the "simplified workflow" mentioned in the [oneMKL documentation](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2024-2/oneapi-mkl-sparse-matmat.html). Could you please confirm that?

Reply:
Yes. We are using the "simplified workflow" of matmat.